src/chain/chain.rs

```diff
@@ -444,33 +464,59 @@ impl Chain {
     pub fn add_member(&mut self, p2p_node: P2pNode, age: u8) {
         self.assert_no_prefix_change("add member");
 
-        // TODO: switch this to true only when the new member is going to be immediately promoted
-        // to elder.
-        self.churn_in_progress = true;
-
         let pub_id = *p2p_node.public_id();
 
-        // Note: it's important that `increment_age_counters` is called *after* the node is added.
+        match self.state.our_members.entry(pub_id) {
+            Entry::Occupied(mut entry) => {
+                if entry.get().state == MemberState::Left {
+                    // Node rejoining
```

Comment:
Why do we have a node rejoining as part of this PR? Should it not never happen (i.e when relocate even on same section change ID?

Reply:
We don't really do node rejoining yet. The rejoining node will still start as infant. That's why I added the TODO below to remind us to implement proper rejoining.

You are right that this branch is currently never going to be entered, but the behaviour is not changed from before this PR, the code is just more explicit now (previously we did this implicitly via `or_insert_with(...)`).